### ISSUE TITLE
It is a bit slow.

### ISSUE THREAD

**Issue by superboi211** [JERScript/JERScript#6]
JERScript is a bit slow compared to other languages so yeah, try to make it faster.

**Comment by superboi211** [comment]
ok

**Comment by jerbear2008** [comment]
From what I've heard, rust is fast, but less portable without compilation. I've compiled js before, and running JERScript in the browser would be nice.

**Comment by jerbear2008** [comment]
> > From what I've heard, rust is fast, but less portable without compilation. I've compiled js before, and running JERScript in the browser would be nice.
> 
> Running JERScript in the browser? Well, I'd have to webpack it first. Then there are a lot of node.js specific functions: So I'd also have to fix those. And the bugs, JERScript on npx is designed to be a CLI rather than a web app. But maybe

Fair point, my point is just that node is very portable. Also, see Browserify for making node work in the browser.

**Comment by Just-a-Unity-Dev** [comment]
> > > From what I've heard, rust is fast, but less portable without compilation. I've compiled js before, and running JERScript in the browser would be nice.
> > 
> > 
> > Running JERScript in the browser? Well, I'd have to webpack it first. Then there are a lot of node.js specific functions: So I'd also have to fix those. And the bugs, JERScript on npx is designed to be a CLI rather than a web app. But maybe
> 
> Fair point, my point is just that node is very portable. Also, see Browserify for making node work in the browser. It can convert node js functions to browser equivalents and combine it with its dependencies.

That's true, but I'm more focused on finishing this lang than porting it to something like the browser, or Rust. I'll probably make a discord bot which runs JERScript in Discord

**Comment by jerbear2008** [comment]
Y

> > > > From what I've heard, rust is fast, but less portable without compilation. I've compiled js before, and running JERScript in the browser would be nice.
> > > 
> > > 
> > > Running JERScript in the browser? Well, I'd have to webpack it first. Then there are a lot of node.js specific functions: So I'd also have to fix those. And the bugs, JERScript on npx is designed to be a CLI rather than a web app. But maybe
> > 
> > 
> > Fair point, my point is just that node is very portable. Also, see Browserify for making node work in the browser. It can convert node js functions to browser equivalents and combine it with its dependencies.
> 
> That's true, but I'm more focused on finishing this lang than porting it to something like the browser, or Rust. I'll probably make a discord bot which runs JERScript in Discord

Yep, finishing takes first priority. Then probably bug fixes and optimizations before work on compilation and more complex stuff.
